File: _non_overlapping_intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {

    let res = []
    intervals.sort((a,b)=>a[1]-b[1])
    res.push(intervals[0])

    intervals.shift()

    let count =0;

    for(let cur of intervals){
        let curStart = cur[0]
        // let curEnd = cur[1]
        let prevEnd = res[res.length-1][1]

        if(curStart<prevEnd){
            //skip
            count++;
        }
        else{
            res.push(cur)
        }
    }

    return count;
    
};

File: Python/non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        si = sorted(intervals, key=lambda item:item[1])

        res = 0
        i=1
        while(i<len(si)):
            prev = si[i-1]
            cur = si[i]

            if(prev[1]>cur[0]):
                res = res+1
                si.pop(i)
                i=i-1
            
            i=i+1

        return res
        
